src/batchUploader.ts

```diff
@@ -129,21 +131,68 @@ export class BatchUploader {
         return offlineStoragePercentage >= rampNumber;
     }
 
+    private shouldDebounceAST(): boolean {
+        const now = Date.now();
+        if (now - this.lastASTEventTime < this.AST_DEBOUNCE_MS) {
+            return true;
+        }
+
+        this.lastASTEventTime = now;
+
+        return false;
+    }
+
+    private createBackgroundASTEvent(): SDKEvent {
+        const now = Date.now();
+        const { _Store } = this.mpInstance;
+        return {
+            EventDataType: MessageType.AppStateTransition,
+            EventName: 'Application Exit',
+            EventCategory: EventType.Other,
+            Timestamp: now,
+            SessionId: _Store.sessionId,
+            MPID: this.mpInstance.Identity.getCurrentUser()?.getMPID(),
+            DeviceId: _Store.deviceId,
+            IsFirstRun: false,
+            SourceMessageId: '',
+            SDKVersion: Constants.sdkVersion,
+            CustomFlags: {},
+            UserAttributes: {},
+            UserIdentities: [],
+            SessionStartDate: _Store.sessionStartDate?.getTime() || now,
+            Debug: _Store.SDKConfig.isDevelopmentMode,
+            CurrencyCode: null,
+            ExpandedEventCount: 0,
+            ActiveTimeOnSite: this.mpInstance._timeOnSiteTimer?.getTimeInForeground() || 0,
+            IsBackgroundAST: true
+        };
+    }
+
     // Adds listeners to be used trigger Navigator.sendBeacon if the browser
     // loses focus for any reason, such as closing browser tab or minimizing window
     private addEventListeners() {
         const _this = this;
 
+        const handleExit = () => {
+            // Check for debounce before creating and queueing event
+            if (_this.shouldDebounceAST()) {
+                return;
+            }
```

Comment:
Based on my previous suggestion, to avoid the side effect, I would do this:

```suggestion
            if (_this.shouldDebounceAST()) {
                this.lastASTEventTime = new Date.now();
                return;
            }
```